cloudmarker/events/azvmextensionevent.py

```diff
@@ -0,0 +1,224 @@
+"""Microsoft Azure VM extension event.
+
+This module defines the :class:`AzVMExtensionEvent` class that
+evaluates Azure VM extensions. This plugin works on the virtual
+machine properties found in the ``ext`` bucket of ``vm_instance_view``
+records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class AzVMExtensionEvent:
+    """Az VM Data extension event plugin."""
+
+    def __init__(self, _whitelisted=None, _blacklisted=None, _required=None):
+        """Create an instance of :class:`AzVMExtensionEvent`.
+
+        Arguments:
+            _whitelisted (list): List of whitelisted extensions.
+            _blacklisted (list): List of blacklisted extensions.
+            _required (list): List of required extensions.
```

Comment:
These can be non-underscore variables, i.e., users can override these variables in their configuration if they wish to.

Reply:
Thanks for your review suggestion @susam . I have addressed this.